src-tauri/src/note/note_service.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{
+    http::{HttpClient, HttpClientImpl},
+    service::Service,
+};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use tauri::Wry;
+use tauri_plugin_store::Store;
+
+use super::Note;
+
+pub type NoteService = dyn Service<Note>;
+
+pub struct LocalNoteService {
+    store: Store<Wry>,
+}
+
+impl LocalNoteService {
+    pub fn new(store: Store<Wry>) -> Self {
+        Self { store }
+    }
+}
+
+#[async_trait]
+impl Service<Note> for LocalNoteService {
+    async fn index(&self) -> Result<Vec<Note>> {
+        let values: Vec<Note> = self
+            .store
+            .values()
+            .iter()
+            .map(|value| serde_json::from_value::<Note>(value.to_owned()).map_err(|e| anyhow!(e)))
+            .collect::<Result<Vec<Note>, _>>()?;
+
+        Ok(values)
+    }
+
+    async fn show(&self, id: u32) -> Result<Note> {
+        match self.store.get(id.to_string()) {
+            Some(note) => Ok(serde_json::from_value(note)?),
+            None => Err(anyhow!("Note {} not found", id)),
+        }
+    }
+
+    async fn store(&self, param: Note) -> Result<()> {
+        self.store
+            .set(param.id.to_string(), serde_json::to_string(&param)?);
+
+        Ok(())
+    }
+
+    async fn update(&self, param: Note) -> Result<()> {
+        self.store
+            .set(param.id.to_string(), serde_json::to_string(&param)?);
+
+        Ok(())
+    }
+
+    async fn destroy(&self, id: u32) -> Result<()> {
+        match self.store.delete(id.to_string()) {
+            true => Ok(()),
+            false => Err(anyhow!("Could not delete note {}", id)),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct RemoteNoteService {
+    client: HttpClientImpl, // TODO: use HttpClient trait
+    slug: String,
+}
+
+impl RemoteNoteService {
+    pub fn new(slug: String, client: HttpClientImpl) -> Self {
+        Self { client, slug }
+    }
+}
```

Comment:
By having slug as a parameter, you're allowing anybody to give a wrong slug on initialization of this service. The slug shouldn't be given to the RemoteNoteService, maybe use .env equivalent, or a config file, and RemoteNoteService should be importing the slug, not receiving it.

Reply:
Yes, the whole configuration part will be done in a separate issue.

Reply:
Fair enough.